snuba/subscriptions/executor.py

```diff
@@ -35,29 +33,30 @@ def __init__(
         self.__metrics = metrics
         self.__concurrent_gauge = Gauge(self.__metrics, "executor.concurrent")
 
-    def __run_query(
-        self, scheduled_at: datetime, request: Request, timer: Timer
-    ) -> Result:
+    def __execute(self, task: ScheduledTask[Subscription], tick: Tick) -> Result:
+        # Measure the amount of time that took between this task being
+        # scheduled and it beginning to execute.
         self.__metrics.timing(
-            "executor.latency", (time.time() - scheduled_at.timestamp()) * 1000,
+            "executor.latency", (time.time() - task.timestamp.timestamp()) * 1000
         )
+
+        # XXX: The ``query`` name is taken from the web views so that all query
+        # performance metrics are reported to the same spot, regardless of
+        # execution environment.
+        timer = Timer("query")
+
+        request = task.task.data.build_request(
+            self.__dataset, task.timestamp, tick.offsets.upper, timer,
+        )
+
         with self.__concurrent_gauge:
+            # XXX: The ``extra`` is discarded from ``RawQueryResult`` since it
+            # is not particularly useful in this context and duplicates data
+            # that is already being published to the query log.
             return parse_and_run_query(self.__dataset, request, timer).result
 
     def execute(self, task: ScheduledTask[Subscription], tick: Tick) -> Future[Result]:
-        timer = Timer("query")
-        try:
-            request = task.task.data.build_request(
-                self.__dataset, task.timestamp, tick.offsets.upper, timer,
-            )
```

Comment:
The original intention behind this was that invalid queries/requests would be caught as fast as possible and fail-fast the consumer. This didn't really work since the types of malformed queries that we can detect before attempting to execute the query is quite limited, and the subscription consumer as designed is unable to detect a failed query until `flush_batch` occurs (at which point all work has been scheduled anyway.)

Reply:
fyi, still makes sense, though when enforcing the AST is valid during parsing we will be able to catch more of those in this step.

Reply:
> fyi, still makes sense, though when enforcing the AST is valid during parsing we will be able to catch more of those in this step.

That is my hope. 🙂